src/Storage/Commands.Storage/Blob/Cmdlet/GetAzureStorageContainer.cs

```diff
@@ -233,11 +240,11 @@ internal void PackCloudBlobContainerWithAcl(IEnumerable<Tuple<CloudBlobContainer
             }
             catch (StorageException e)
             {
-                if (!e.IsNotFoundException())
+                if (!e.IsNotFoundException() && !e.IsFordiddenException())
```

Comment:
@blueww we should change how exceptions are being thrown here so that the properties of the exception are preserved. This means only catching the exceptions that throw a 403 or 404. We should also put the warning you're adding above in here rather than having it before the check. You can change the message slightly so it applies to both 403 and 404 scenarios. It should look something like the following:

```cs
catch (StorageException e) when (e.IsNotFoundException() || e.IsForbiddenException())
{
    WriteWarning("Denied permission to query the permission of the specified container.");
}
```

Reply:
@cormacpayne 

If change as you required, will get following error, this is because query container ACL is executed in aync task. So we can't write warning in it. Any idea？
BTW，I have modify the catch exception as you required.

>>Get-AzureStorageContainer : The WriteObject and WriteError methods cannot be called from outside the overrides of the BeginProcessing, ProcessRecord, and EndProcessing methods, and they can only be called from within the same thread. Validate 
that the cmdlet makes these calls correctly, or contact Microsoft Customer Support Services.
At line:1 char:1
>>+ Get-AzureStorageContainer -Context $ctxsas
>>+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>    + CategoryInfo          : CloseError: (:) [Get-AzureStorageContainer], PSInvalidOperationException
>>    + FullyQualifiedErrorId : PSInvalidOperationException,Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet.GetAzureStorageContainerCommand